### ISSUE TITLE
Problem with System.http. not found at runtime

### ISSUE THREAD

**Issue by rgomez90** [PromoFaux/Matterhook.NET.MatterhookClient#3]
I wanted to use this dll to notify in Mattermost events of my app (.NET 4.6).

Problem is when trying to create an instance of Matterhook client, I get the exception, that System.Net.Http 4.0.0 was not found. 

I tried updating packages, rebuilding, but nothing. I can't find the dll in the project's bin folder.

If I manually copy it to the bin folder, an exception is no more thrown and my app works like it should.

I am doing sth wrong or is this a bug?

**Comment by PromoFaux** [comment]
Hmmm, did you build from source or use the nuget package?

https://www.nuget.org/packages/Matterhook.NET.MatterhookClient/1.2.4

**Comment by PromoFaux** [comment]
@rgomez90 did the nuget package work for you?

**Comment by rgomez90** [comment]
I used the NuGet Package (latest Release)

**Comment by PromoFaux** [comment]
Sorry, have been quite busy... will investigate now :)

**Comment by PromoFaux** [comment]
Odd. I just created a new console app, added `Matterhook.NET.MatterhookClient` via `Nuget` and am able to implement it, and build with no errors:

![image](https://user-images.githubusercontent.com/1998970/41201033-845e8d68-6ca8-11e8-9497-9ad781d55e48.png)

Is the source for your project you're attempting to use with it available on Github?

**Comment by PromoFaux** [comment]
@rgomez90 Did you manage to solve the issue?

**Comment by rgomez90** [comment]
@PromoFaux Thank for your response and apologize for the late answer. 
The app is a small measurement app for my company, so I'm not allowed to show the source code. 
I suppose then, it is some issue specific of my solution, but I still haven't found why (I haven't researched a lot, since I get it working by manually copying the DLL file to the output folder, but I will take the time to investigate)

**Comment by rgomez90** [comment]
@PromoFaux I`ve seen this lines in my app.config and have thought it may can have sth. to do with my issue?
![image](https://user-images.githubusercontent.com/12009909/41732427-4d8644d4-7581-11e8-931a-99091af29f08.png)

**Comment by rgomez90** [comment]
So I have the reason.

My solution has a `UI `Project (Main project) which references and uses a DLL project called `BLL`.
`BLL` has this library installed as NuGet package. `BLL` exposed a class `MattermostService` with a method `PostAsync`. 

I was cathing an `HttpRequestException` in the presenter class of the view, which uses the `MattermostService` (MVC-WinForms). Doing this a reference to `System.Net.Http` was added to the presenter class and at runtime I got an exception of file not found.

I look at `CopyLocal` in the dependence of `UI`-project and set it to true (was set to False). Then I realize the DLL being copied was not the same version  as the expected by Matterhook.

Now I don't catch any `HttpRequestException` in my presenter and now I'm cathing a custom one. The reference to `.Net.Http` in my presenter is gone, and now the right DLL is being copied.

I think, that was the cause and hope you understand, what was my issue and my solution to it.
